kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+                	 if (parsingError.size() > 0) {
+                		 //parsing error, write error and continue	   
+                		 reportWriterService.writeError(originEntry, parsingError);
+                		 createOutputEntry(GLEN_RECORD, OUTPUT_ERR_FILE_ps);
+                		 reportBudgetAdjustDocErrors++;
+                		 continue;
+                     }
+                     
+                     if (isBudgetAdjustmentTransaction(originEntry)) {
+                    	 
+                    	//check to make sure that Account has Income Stream information
+                    	 Account baAccount = originEntry.getAccount();    	 
+                         if (ObjectUtils.isNotNull(baAccount)) {                        
+                        	 if (ObjectUtils.isNull(baAccount.getIncomeStreamFinancialCoaCode()) || ObjectUtils.isNull(baAccount.getIncomeStreamAccountNumber())) {             	
+	                        	 //no income stream info, write error and continue	                        	
+	                             Message errorMsg = new Message("No Account Income Stream information found for this record.", Message.TYPE_FATAL);
+	                             reportWriterService.writeError(originEntry, errorMsg);
+	                             createOutputEntry(GLEN_RECORD, OUTPUT_ERR_FILE_ps);
+	                             reportBudgetAdjustDocErrors++;
+	                             continue;
+	                         }
+                         }
+                         
+                         //check to make sure that Object Type Code is present (BO input may not have this)
+                         if (StringUtils.isBlank(originEntry.getFinancialObjectTypeCode())) {
+                        	 String financialObjectTypeCode = getFinancialObjectTypeCode(originEntry.getUniversityFiscalYear(), originEntry.getChartOfAccountsCode(), originEntry.getFinancialObjectCode());
+                        	 if (ObjectUtils.isNull(financialObjectTypeCode)) {
+                        		//no valid Object Code info, write error and continue	                        	
+	                             Message errorMsg = new Message("No Object Code information found for this record.", Message.TYPE_FATAL);
+	                             reportWriterService.writeError(originEntry, errorMsg);
+	                             createOutputEntry(GLEN_RECORD, OUTPUT_ERR_FILE_ps);
+	                             reportBudgetAdjustDocErrors++;
+	                             continue;
+                        	 }
+                        	 else {
+                        		 originEntry.setFinancialObjectTypeCode(financialObjectTypeCode);
+                        	 }
+                         }
+                         
+                         // Make sure the row will be unique when adding to the budget adjustment table by adjusting the transaction sequence id
+                         if (originEntry.getDocumentNumber().equals(saveDocNumber)) {
```

Comment:
Where is  saveDocNumber assigned? The only previous mention of this var is in:
 String saveDocNumber = new String("");